lib/onEvent/message.js

```diff
@@ -2,35 +2,81 @@ const Kirbi = require('../../../../kirbi');
 
 //initialize AI
 if (Kirbi.Config.elizaEnabled && !Kirbi.Eliza) {
-	var Eliza = Kirbi.require('./extras/eliza');
+	let Eliza = Kirbi.require('./extras/eliza');
 	Kirbi.Eliza = new Eliza();
 	console.log('Eliza enabled.');
 	Kirbi.Eliza.memSize = 500;
 };
 
+function currentChannelHasHook (channelId) {
+	if (Kirbi.Auth.slack.webhooks[Kirbi.Slack.dataStore.getChannelById(channelId).name]) return true;
+	return false;
+};
+
 function slackMessageCb (output, msg, expires, delCalling) {
 	//we need to format the message output for the Slack API first...
-	let reformatted = {};
+	if (typeof output === 'object') {
+		let reformatted;
+		reformatted = {};
+		reformatted.attachments = [];
+		reformatted.attachments[0] = {};
+		reformatted.attachments[0].fallback = 'There was an error';
+		if (output.embed.color) reformatted.attachments[0].color = `#${output.embed.color}`;
+		if (output.embed.title) reformatted.attachments[0].title = output.embed.title;
+		if (output.embed.description) reformatted.attachments[0].text = output.embed.description;
+		if (typeof output.embed.image !== 'undefined') reformatted.attachments[0].image_url = output.embed.image.url;
+		if (typeof output.embed.thumbnail !== 'undefined') reformatted.attachments[0].thumb_url = output.embed.thumbnail.url;
+		if (output.embed.footer) reformatted.attachments[0].footer = output.embed.footer;
```

Comment:
I'd group all the attributes that aren't being changed together after you first make the attachment.

Reply:
they aren't present in all, I'll just leave the check in